.github/workflows/Release.yml

```diff
@@ -0,0 +1,82 @@
+name: "Publish Release"
+on:
+  workflow_dispatch:
+env:
+  build_dir: ./build
+  tarballs_dir: ./build/tarballs
+jobs:
+  release:
+    name: "Publish Pre-Release"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Needed determine latest tag
+      - name: Get Latest Tag
+        id: latest_tag
+        uses: WyriHaximus/github-action-get-previous-tag@v1
+        # TODO: in future we might need to perform backports, at which point we'll need to check
+        # that the project.version has not already been tagged 
+      - name: Check for updated Project version
+        id: project_toml
+        shell: julia --color=yes --project {0}
+        env:
+          LATEST_TAG: ${{ steps.latest_tag.outputs.tag }}
+        run: |
+          using Pkg.Types
+          project = read_project("Project.toml")
+          latest_tag = parse(VersionNumber, ENV["LATEST_TAG"])
+          @info "Project version: project.version"
+          @info "Latest tag: latest_tag"
+          project.version > latest_tag || exit(1)  # error if can't make a tag
+          open(ENV["GITHUB_OUTPUT"], "a") do io
+              println(io, "version=$(project.version)")
+              println(io, "tag=v$(project.version)")
+          end
+      - uses: actions/checkout@v4
+      - name: Download artifacts
+        id: download-artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          branch: main
+          name: ray_julia_libraries
+          path: ${{ env.tarballs_dir }}
+          workflow: CI.yml
```

Comment:
does this run any risk of multiple PRs being merged and picking up either too old or too new artifacts?

Reply:
actually yeah I think that's a risk if we use `branch` https://github.com/dawidd6/action-download-artifact/issues/240

but if we use `commit` it should be deterministic

Reply:
I believe there is a definite danger here of making a release with the wrong tarballs